flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV120.java

```diff
@@ -168,4 +174,47 @@ public CatalogColumnStatisticsDataDate toFlinkDateColStats(ColumnStatisticsData
 			throw new CatalogException("Failed to create Flink statistics for date column", e);
 		}
 	}
+
+	@Override
+	public Set<String> listBuiltInFunctions() {
+		try {
+			Method method = FunctionRegistry.class.getMethod("getFunctionNames");
+			// makeSpecFromName is a static method
+			Set<String> names = (Set<String>) method.invoke(null);
+
+			return names.stream()
+				.filter(n -> getFunctionInfo(n).isBuiltIn())
+				.collect(Collectors.toSet());
+		} catch (Exception ex) {
+			throw new CatalogException("Failed to invoke Warehouse.makeSpecFromName()", ex);
+		}
+	}
+
+	private FunctionInfo getFunctionInfo(String name) {
+		try {
+			return FunctionRegistry.getFunctionInfo(name);
+		} catch (SemanticException e) {
+			throw new FlinkHiveException(
+				String.format("Failed getting function info for %s", name), e);
+		}
+	}
+
+	@Override
+	public FunctionInfo getBuiltInFunctionInfo(String name) {
+		FunctionInfo functionInfo;
+
+		try {
+			functionInfo = FunctionRegistry.getFunctionInfo(name);
+		} catch (SemanticException e) {
+			throw new FlinkHiveException(
+				String.format("Failed getting function info for %s", name), e);
+		}
+
+		if (functionInfo.isBuiltIn()) {
+			return functionInfo;
+		} else {
+			throw new FlinkHiveUDFException(
```

Comment:
We may add this to the API to be clear about what happens if the function is not a built-in function